include/eve/detail/top_bits.hpp

```diff
@@ -92,7 +111,7 @@ struct top_bits
   static constexpr bool is_avx512_logical = !abi_type_t<Logical>::is_wide_logical;
   static constexpr std::ptrdiff_t static_size = Logical::static_size;
   static constexpr std::ptrdiff_t bits_per_element =
-    sizeof(element_type_t<Logical>) == 2 ? 2 : 1;
+    sizeof(element_type_t<Logical>) == 2 && !is_avx512_logical ? 2 : 1;
```

Comment:
You compute is_avx512_logical as ` !abi_type_t<Logical>::is_wide_logical` then you re apply ! in the ternary condition.
Is it correct or is one of those ! useless ?